source/shared/core_stmt.cpp

```diff
@@ -2004,6 +2019,19 @@ void default_sql_size_and_scale( _Inout_ sqlsrv_stmt* stmt, _In_opt_ unsigned in
     }
 }
 
+void ae_get_sql_type_info( _Inout_ sqlsrv_stmt* stmt, _In_opt_ SQLULEN paramno, _In_ SQLSMALLINT direction, _In_ zval* param_z, _In_ SQLSRV_ENCODING encoding,
+                           _Out_ SQLSMALLINT& sql_type, _Out_ SQLULEN& column_size, _Out_ SQLSMALLINT& decimal_digits TSRMLS_DC )
+{
+    SQLSMALLINT Nullable;
+    core::SQLDescribeParam( stmt, paramno + 1, &sql_type, &column_size, &decimal_digits, &Nullable );
+    // without AE, binding an INPUT bigint with sql type SQL_WVARCHAR returns a conversion error between nvarchar and bigint
+    // similate without AE by setting the sql type back to SQL_WVARCHAR using default_sql_type
+    if ( sql_type == SQL_BIGINT && Z_TYPE_P(param_z) == IS_STRING && direction == SQL_PARAM_INPUT ) {
```

Comment:
I don't know how I missed this before. Anyway, the comment here is confusing... isn't the prerequisite of this method that AE is enabled? 
Also, why is `default_sql_size_and_scale` called to reset the column size? Besides, what if type of param_z is not a string?
Does this problem happen only with input param? Why not output or in/out param? 
Have you tested when encoding is SQLSRV_ENCODING_CHAR?

Reply:
Yes this function is only called when AE is enabled.

default_sql_size_and_scale is called to reset the column size because we want to mimic what happens without AE (i.e., as explained in the comment, without AE, there is an error in the conversion between SQL_WVARCHAR and SQL_BIGINT). If this block of code is not here, the will no longer be converting between bigint to wvarchar, instead it'll be converting between bigint and bigint (which is set by SQLDescribeParam). Everything would be fine at this point, except it will no longer mimic the behavior of without AE.

If the type of param_z is not a string, then we use the sql_type returned by SQLDescribeParam.

This is only for input params. I've tested without AE with output and in/out bigint, the conversion problem does not exist.

Yes, I've tested with SQLSRV_ENCODING_CHAR